scripts/test_docs_code_blocks.py

```
@@ -0,0 +1,486 @@
+
+# A `<placeholder>` template (e.g. `store <name> as <value>`) is documentation
+# pseudo-syntax, not runnable WFL — recognize it so we skip execution instead of
+# running it and counting a guaranteed lexing failure as a hard error.
+PLACEHOLDER_RE = re.compile(r"<[^>\n]{1,40}>")
+
+# WFL's built-in test framework (describe/test/expect) only runs under `--test`.
+TEST_FRAMEWORK_RE = re.compile(r"^\s*(describe|test)\s+\"|^\s*expect\s", re.MULTILINE)
+
+
+def looks_like_snippet(code: str) -> bool:
+    stripped = [l for l in code.splitlines() if l.strip()]
```

Ruff (0.15.20)

[error] Ambiguous variable name: `l`

(E741)